diff --git a/common/math/interval.h b/common/math/interval.h
index 310add212..81d8c453e 100644
--- a/common/math/interval.h
+++ b/common/math/interval.h
@@ -70,10 +70,12 @@ namespace embree
       __forceinline friend const Interval intersect( const Interval& a, const Interval& b ) { return Interval(max(a.lower, b.lower), min(a.upper, b.upper)); }
       __forceinline friend const Interval intersect( const Interval& a, const Interval& b, const Interval& c ) { return intersect(a,intersect(b,c)); }
       __forceinline friend const Interval intersect( const Interval& a, const Interval& b, const Interval& c, const Interval& d ) { return intersect(intersect(a,b),intersect(c,d)); }       
-      
+
+      #if !defined(__APPLE__)
       friend embree_ostream operator<<(embree_ostream cout, const Interval& a) {
         return cout << "[" << a.lower << ", " << a.upper << "]";
       }
+      #endif
       
       ////////////////////////////////////////////////////////////////////////////////
       /// Constants
diff --git a/common/math/obbox.h b/common/math/obbox.h
index 2fe8bbf07..b27ba7a49 100644
--- a/common/math/obbox.h
+++ b/common/math/obbox.h
@@ -25,9 +25,11 @@ namespace embree
     __forceinline OBBox (const LinearSpace3<T>& space, const BBox<T>& bounds) 
       : space(space), bounds(bounds) {}
     
+    #if !defined(__APPLE__)
     friend embree_ostream operator<<(embree_ostream cout, const OBBox& p) {
       return cout << "{ space = " << p.space << ", bounds = " << p.bounds << "}";
     }
+    #endif
     
   public:
     LinearSpace3<T> space; //!< orthonormal transformation
diff --git a/common/math/range.h b/common/math/range.h
index 909fadb99..d65908c34 100644
--- a/common/math/range.h
+++ b/common/math/range.h
@@ -74,9 +74,11 @@ namespace embree
         return r0.size() < r1.size();
       }
 	
+      #if !defined(__APPLE__)
       friend embree_ostream operator<<(embree_ostream cout, const range& r) {
         return cout << "range [" << r.begin() << ", " << r.end() << "]";
       }
+      #endif
       
       Ty _begin, _end;
     };
@@ -128,9 +130,11 @@ namespace embree
         _ext_end += plus;
       }
 
+      #if !defined(__APPLE__)
       friend embree_ostream operator<<(embree_ostream cout, const extended_range& r) {
         return cout << "extended_range [" << r.begin() << ", " << r.end() <<  " (" << r.ext_end() << ")]";
       }
+      #endif
       
       Ty _ext_end;
     };
diff --git a/kernels/builders/heuristic_binning.h b/kernels/builders/heuristic_binning.h
index 41be6183b..950fdf627 100644
--- a/kernels/builders/heuristic_binning.h
+++ b/kernels/builders/heuristic_binning.h
@@ -99,10 +99,12 @@ namespace embree
           return scale[dim] == 0.0f;
         }
         
+        #if !defined(__APPLE__)
         /*! stream output */
         friend embree_ostream operator<<(embree_ostream cout, const BinMapping& mapping) {
           return cout << "BinMapping { num = " << mapping.num << ", ofs = " << mapping.ofs << ", scale = " << mapping.scale << "}";
         }
+        #endif
         
       public:
         size_t num;
@@ -140,9 +142,11 @@ namespace embree
         __forceinline float splitSAH() const { return sah; }
         
         /*! stream output */
+        #if !defined(__APPLE__)
         friend embree_ostream operator<<(embree_ostream cout, const BinSplit& split) {
           return cout << "BinSplit { sah = " << split.sah << ", dim = " << split.dim << ", pos = " << split.pos << "}";
         }
+        #endif
         
       public:
         float sah;                //!< SAH cost of the split
diff --git a/kernels/builders/heuristic_spatial.h b/kernels/builders/heuristic_spatial.h
index 8b3499ac8..839db11e2 100644
--- a/kernels/builders/heuristic_spatial.h
+++ b/kernels/builders/heuristic_spatial.h
@@ -100,9 +100,11 @@ namespace embree
         __forceinline float splitSAH() const { return sah; }
         
         /*! stream output */
+        #if !defined(__APPLE__)
         friend embree_ostream operator<<(embree_ostream cout, const SpatialBinSplit& split) {
           return cout << "SpatialBinSplit { sah = " << split.sah << ", dim = " << split.dim << ", pos = " << split.pos << ", left = " << split.left << ", right = " << split.right << ", factor = " << split.factor << "}";
         }
+        #endif
         
       public:
         float sah;                 //!< SAH cost of the split
diff --git a/kernels/builders/priminfo.h b/kernels/builders/priminfo.h
index fee515247..d4ca2d61b 100644
--- a/kernels/builders/priminfo.h
+++ b/kernels/builders/priminfo.h
@@ -148,9 +148,11 @@ namespace embree
       }
       
       /*! stream output */
+      #if !defined(__APPLE__)
       friend embree_ostream operator<<(embree_ostream cout, const PrimInfoT& pinfo) {
 	return cout << "PrimInfo { begin = " << pinfo.begin << ", end = " << pinfo.end << ", geomBounds = " << pinfo.geomBounds << ", centBounds = " << pinfo.centBounds << "}";
       }
+      #endif
       
     public:
       size_t begin,end;          //!< number of primitives
@@ -240,6 +242,7 @@ namespace embree
       }
       
       /*! stream output */
+      #if !defined(__APPLE__)
       friend embree_ostream operator<<(embree_ostream cout, const PrimInfoMBT& pinfo) 
       {
 	return cout << "PrimInfo { " << 
@@ -250,6 +253,7 @@ namespace embree
           ", centBounds = " << pinfo.centBounds << 
           "}";
       }
+      #endif
       
     public:
       range<size_t> object_range; //!< primitive range
diff --git a/kernels/bvh/bvh_node_aabb.h b/kernels/bvh/bvh_node_aabb.h
index 3fd9fc7d1..dc7e8e82d 100644
--- a/kernels/bvh/bvh_node_aabb.h
+++ b/kernels/bvh/bvh_node_aabb.h
@@ -196,6 +196,7 @@ namespace embree
     __forceinline const NodeRef& child(size_t i) const { assert(i<N); return children[i]; }
     
     /*! output operator */
+    #if !defined(__APPLE__)
     friend embree_ostream operator<<(embree_ostream o, const AABBNode_t& n)
     {
       o << "AABBNode { " << embree_endl;
@@ -211,6 +212,7 @@ namespace embree
       o << "}" << embree_endl;
       return o;
     }
+    #endif
     
   public:
     vfloat<N> lower_x;           //!< X dimension of lower bounds of all N children.
diff --git a/kernels/bvh/bvh_node_aabb_mb.h b/kernels/bvh/bvh_node_aabb_mb.h
index 001f526c2..b436580e6 100644
--- a/kernels/bvh/bvh_node_aabb_mb.h
+++ b/kernels/bvh/bvh_node_aabb_mb.h
@@ -221,6 +221,7 @@ namespace embree
     __forceinline const NodeRef& child(size_t i) const { assert(i<N); return children[i]; }
     
     /*! stream output operator */
+    #if !defined(__APPLE__)
     friend embree_ostream operator<<(embree_ostream cout, const AABBNodeMB_t& n) 
     {
       cout << "AABBNodeMB {" << embree_endl;
@@ -236,6 +237,7 @@ namespace embree
       cout << "}";
       return cout;
     }
+    #endif
     
   public:
     vfloat<N> lower_x;        //!< X dimension of lower bounds of all N children.
diff --git a/kernels/bvh/bvh_node_aabb_mb4d.h b/kernels/bvh/bvh_node_aabb_mb4d.h
index 3b966fd05..6b946e50c 100644
--- a/kernels/bvh/bvh_node_aabb_mb4d.h
+++ b/kernels/bvh/bvh_node_aabb_mb4d.h
@@ -91,6 +91,7 @@ namespace embree
     }
     
     /*! stream output operator */
+    #if !defined(__APPLE__)
     friend embree_ostream operator<<(embree_ostream cout, const AABBNodeMB4D_t& n) 
     {
       cout << "AABBNodeMB4D {" << embree_endl;
@@ -107,6 +108,7 @@ namespace embree
       cout << "}";
       return cout;
     }
+    #endif
     
   public:
     vfloat<N> lower_t;        //!< time dimension of lower bounds of all N children
diff --git a/kernels/bvh/bvh_node_obb.h b/kernels/bvh/bvh_node_obb.h
index e6b500691..770b9a8b5 100644
--- a/kernels/bvh/bvh_node_obb.h
+++ b/kernels/bvh/bvh_node_obb.h
@@ -86,11 +86,13 @@ namespace embree
     __forceinline const NodeRef& child(size_t i) const { assert(i<N); return children[i]; }
     
     /*! output operator */
+    #if !defined(__APPLE__)
     friend embree_ostream operator<<(embree_ostream o, const OBBNode_t& n)
     {
       o << "UnAABBNode { " << n.naabb << " } " << embree_endl;
       return o;
     }
+    #endif
     
   public:
     AffineSpace3vf<N> naabb;   //!< non-axis aligned bounding boxes (bounds are [0,1] in specified space)
diff --git a/kernels/bvh/bvh_node_qaabb.h b/kernels/bvh/bvh_node_qaabb.h
index 99671ddc5..b1c87756e 100644
--- a/kernels/bvh/bvh_node_qaabb.h
+++ b/kernels/bvh/bvh_node_qaabb.h
@@ -132,6 +132,7 @@ namespace embree
     Vec3f start;
     Vec3f scale;
     
+    #if !defined(__APPLE__)
     friend embree_ostream operator<<(embree_ostream o, const QuantizedBaseNode_t& n)
     {
       o << "QuantizedBaseNode { " << embree_endl;
@@ -146,6 +147,7 @@ namespace embree
       o << "}" << embree_endl;
       return o;
     }
+    #endif
     
   };
 
diff --git a/kernels/common/scene_grid_mesh.h b/kernels/common/scene_grid_mesh.h
index fb6fed445..61078d370 100644
--- a/kernels/common/scene_grid_mesh.h
+++ b/kernels/common/scene_grid_mesh.h
@@ -34,9 +34,11 @@ namespace embree
       }
 
       /*! outputs grid structure */
+      #if !defined(__APPLE__)
       __forceinline friend embree_ostream operator<<(embree_ostream cout, const Grid& t) {
         return cout << "Grid { startVtxID " << t.startVtxID << ", lineVtxOffset " << t.lineVtxOffset << ", resX " << t.resX << ", resY " << t.resY << " }";
       }
+      #endif
     };
 
   public:
diff --git a/kernels/common/scene_quad_mesh.h b/kernels/common/scene_quad_mesh.h
index bd8eeaaeb..296d6c2f3 100644
--- a/kernels/common/scene_quad_mesh.h
+++ b/kernels/common/scene_quad_mesh.h
@@ -20,9 +20,11 @@ namespace embree
       uint32_t v[4];
 
       /*! outputs triangle indices */
+      #if !defined(__APPLE__)
       __forceinline friend embree_ostream operator<<(embree_ostream cout, const Quad& q) {
         return cout << "Quad {" << q.v[0] << ", " << q.v[1] << ", " << q.v[2] << ", " << q.v[3] << " }";
       }
+      #endif
     };
 
   public:
diff --git a/kernels/common/scene_triangle_mesh.h b/kernels/common/scene_triangle_mesh.h
index ad3f602fd..3524466cd 100644
--- a/kernels/common/scene_triangle_mesh.h
+++ b/kernels/common/scene_triangle_mesh.h
@@ -20,9 +20,11 @@ namespace embree
       uint32_t v[3];
 
       /*! outputs triangle indices */
+      #if !defined(__APPLE__)
       __forceinline friend embree_ostream operator<<(embree_ostream cout, const Triangle& t) {
         return cout << "Triangle { " << t.v[0] << ", " << t.v[1] << ", " << t.v[2] << " }";
       }
+      #endif
     };
 
   public:
diff --git a/kernels/geometry/pointi.h b/kernels/geometry/pointi.h
index bed04116b..ce49d6565 100644
--- a/kernels/geometry/pointi.h
+++ b/kernels/geometry/pointi.h
@@ -210,10 +210,12 @@ namespace embree
     };
 
     /*! output operator */
+    #if !defined(__APPLE__)
     friend __forceinline embree_ostream operator<<(embree_ostream cout, const PointMi& line)
     {
       return cout << "Line" << M << "i {" << line.v0 << ", " << line.geomID() << ", " << line.primID() << "}";
     }
+    #endif
 
    public:
     unsigned char gtype;
diff --git a/kernels/geometry/quadi.h b/kernels/geometry/quadi.h
index 70a7bdf15..f2a5e4261 100644
--- a/kernels/geometry/quadi.h
+++ b/kernels/geometry/quadi.h
@@ -159,6 +159,7 @@ namespace embree
       return linearBounds(scene, time_range);
     }
 
+    #if !defined(__APPLE__)
     friend embree_ostream operator<<(embree_ostream cout, const QuadMi& quad) {
       return cout << "QuadMi<" << M << ">( "
 #if !defined(EMBREE_COMPACT_POLYS)
@@ -166,6 +167,7 @@ namespace embree
 #endif
                   << "geomID = " << quad.geomIDs << ", primID = " << quad.primIDs << " )";
     }
+    #endif
 
   protected:
 #if !defined(EMBREE_COMPACT_POLYS)
diff --git a/kernels/geometry/subgrid.h b/kernels/geometry/subgrid.h
index ce54421ca..ed5b98333 100644
--- a/kernels/geometry/subgrid.h
+++ b/kernels/geometry/subgrid.h
@@ -281,11 +281,11 @@ namespace embree
           return allBounds;
         }
 
-
+#if !defined(__APPLE__)
         friend embree_ostream operator<<(embree_ostream cout, const SubGrid& sg) {
           return cout << "SubGrid " << " ( x " << sg.x() << ", y = " << sg.y() << ", geomID = " << sg.geomID() << ", primID = " << sg.primID() << " )";
         }
-
+#endif
         __forceinline unsigned int geomID() const { return _geomID; }
         __forceinline unsigned int primID() const { return _primID; }
         __forceinline unsigned int x() const { return (unsigned int)_x & 0x7fff; }
@@ -379,7 +379,7 @@ namespace embree
 
         unsigned int _geomID;    // geometry ID of mesh
 
-
+#if !defined(__APPLE__)
         friend embree_ostream operator<<(embree_ostream cout, const SubGridQBVHN& sg) {
           cout << "SubGridQBVHN " << embree_endl;
           for (size_t i=0;i<N;i++)
@@ -393,7 +393,7 @@ namespace embree
           cout << "upperZ " << sg.qnode.dequantizeUpperZ() << embree_endl;
           return cout;
         }
-
+#endif
       };
 
       template<int N>
@@ -489,7 +489,7 @@ namespace embree
         float time_scale;
         unsigned int _geomID;    // geometry ID of mesh
 
-
+#if !defined(__APPLE__)
         friend embree_ostream operator<<(embree_ostream cout, const SubGridMBQBVHN& sg) {
           cout << "SubGridMBQBVHN " << embree_endl;
           for (size_t i=0;i<N;i++)
@@ -511,7 +511,7 @@ namespace embree
           cout << "upperZ " << sg.qnode.node1.dequantizeUpperZ() << embree_endl;
           return cout;
         }
-
+#endif
       };
 
 }
diff --git a/kernels/subdiv/bezier_curve.h b/kernels/subdiv/bezier_curve.h
index a5adad5cc..473bea82c 100644
--- a/kernels/subdiv/bezier_curve.h
+++ b/kernels/subdiv/bezier_curve.h
@@ -97,9 +97,11 @@ namespace embree
       
       bool hasRoot() const;
       
+      #if !defined(__APPLE__)
       friend embree_ostream operator<<(embree_ostream cout, const LinearBezierCurve& a) {
         return cout << "LinearBezierCurve (" << a.v0 << ", " << a.v1 << ")";
       }
+      #endif
     };
   
   template<> __forceinline bool LinearBezierCurve<Interval1f>::hasRoot() const {
@@ -134,9 +136,11 @@ namespace embree
         return merge(BBox<V>(v0),BBox<V>(v1),BBox<V>(v2));
       }
       
+      #if !defined(__APPLE__)
       friend embree_ostream operator<<(embree_ostream cout, const QuadraticBezierCurve& a) {
         return cout << "QuadraticBezierCurve ( (" << a.u.lower << ", " << a.u.upper << "), " << a.v0 << ", " << a.v1 << ", " << a.v2 << ")";
       }
+      #endif
     };
   
   
diff --git a/kernels/subdiv/gregory_patch.h b/kernels/subdiv/gregory_patch.h
index 9026d5c40..757ad5d9f 100644
--- a/kernels/subdiv/gregory_patch.h
+++ b/kernels/subdiv/gregory_patch.h
@@ -841,6 +841,7 @@ namespace embree
       return bounds;
     }
     
+    #if !defined(__APPLE__)
     friend embree_ostream operator<<(embree_ostream o, const GregoryPatchT& p)
     {
       for (size_t y=0; y<4; y++)
@@ -852,6 +853,7 @@ namespace embree
 	  o << "f[" << y << "][" << x << "] " << p.f[y][x] << embree_endl;
       return o;
     } 
+    #endif
   };
 
   typedef GregoryPatchT<Vec3fa,Vec3fa_t> GregoryPatch3fa;
diff --git a/kernels/subdiv/linear_bezier_patch.h b/kernels/subdiv/linear_bezier_patch.h
index dcdb101d7..a3d48cee6 100644
--- a/kernels/subdiv/linear_bezier_patch.h
+++ b/kernels/subdiv/linear_bezier_patch.h
@@ -231,6 +231,7 @@ namespace embree
           return (R.begin()-L.begin())+(R.end()-L.end());
         }
         
+        #if !defined(__APPLE__)
         friend embree_ostream operator<<(embree_ostream cout, const TensorLinearCubicBezierSurface& a)
         {
           return cout << "TensorLinearCubicBezierSurface" << embree_endl
@@ -239,6 +240,7 @@ namespace embree
                       << "  R = " << a.R << embree_endl
                       << "}";
         }
+        #endif
 
         friend __forceinline TensorLinearCubicBezierSurface clerp(const TensorLinearCubicBezierSurface& a, const TensorLinearCubicBezierSurface& b, const float t) {
           return TensorLinearCubicBezierSurface(clerp(a.L,b.L,V(t)), clerp(a.R,b.R,V(t)));
@@ -386,6 +388,7 @@ namespace embree
         return (R.begin()-L.begin())+(R.end()-L.end());
       }
       
+      #if !defined(__APPLE__)
       friend embree_ostream operator<<(embree_ostream cout, const TensorLinearCubicBezierSurface& a)
       {
         return cout << "TensorLinearCubicBezierSurface" << embree_endl
@@ -394,6 +397,7 @@ namespace embree
                     << "  R = " << a.getR() << embree_endl
                     << "}";
       }
+      #endif
     };
 
     typedef TensorLinearCubicBezierSurface<float> TensorLinearCubicBezierSurface1f;
